{
  "active_lock_reason": null,
  "assignee": null,
  "assignees": [],
  "author_association": "CONTRIBUTOR",
  "body": "**What happened**:\r\n\r\nHousekeeping sync loop finds an orphaned pod and tries to clean it up. The kubepod slice, as seen below, is not removed by `cleanupOrphanedPodCgroups -> go pcm.Destroy(val)` with no indication as to why. This leads to the docker scope timing out and entering a failed state. The returned error(s) from runc are swallowed making debugging difficult.\r\n\r\n```\r\nOct 16 21:18:26 server atomic-openshift-node: I1016 21:18:26.531606    1694 kubelet.go:1907] SyncLoop (housekeeping)\r\nOct 16 21:18:26 server atomic-openshift-node: I1016 21:18:26.670054    1694 kubelet_pods.go:1082] Killing unwanted pod \"test-cron-three-1571285880-bb\"\r\nOct 16 21:18:26 server atomic-openshift-node: I1016 21:18:26.792307    1694 kubelet_pods.go:1682] Orphaned pod \"20f666f1-f095-11e9-916f-0050568379a2\" found, removing pod cgroups\r\nOct 16 21:18:26 server systemd: Stopping libcontainer container 4396d3ff70201d494fa265e1966435288d6806acb99f0d601bf56f8804f79a24.\r\nOct 16 21:19:57 server systemd: docker-4396d3ff70201d494fa265e1966435288d6806acb99f0d601bf56f8804f79a24.scope stopping timed out. Killing.\r\nOct 16 21:19:57 server systemd: Stopped libcontainer container 4396d3ff70201d494fa265e1966435288d6806acb99f0d601bf56f8804f79a24.\r\nOct 16 21:19:57 server systemd: Unit docker-4396d3ff70201d494fa265e1966435288d6806acb99f0d601bf56f8804f79a24.scope entered failed state.\r\n```\r\n\r\n\r\n**What you expected to happen**:\r\nAs per other areas of the sync loop, useful errors are logged at debug level due to the async calls.\r\n\r\n**How to reproduce it (as minimally and precisely as possible)**:\r\nN/A\r\n\r\n**Anything else we need to know?**:\r\nN/A\r\n\r\n**Environment**:\r\n- Kubernetes version (use `kubectl version`): master\r\n",
  "closed_at": "2020-05-17T01:59:37Z",
  "closed_by": {
    "avatar_url": "https://avatars0.githubusercontent.com/u/20407524?v=4",
    "events_url": "https://api.github.com/users/k8s-ci-robot/events{/privacy}",
    "followers_url": "https://api.github.com/users/k8s-ci-robot/followers",
    "following_url": "https://api.github.com/users/k8s-ci-robot/following{/other_user}",
    "gists_url": "https://api.github.com/users/k8s-ci-robot/gists{/gist_id}",
    "gravatar_id": "",
    "html_url": "https://github.com/k8s-ci-robot",
    "id": 20407524,
    "login": "k8s-ci-robot",
    "node_id": "MDQ6VXNlcjIwNDA3NTI0",
    "organizations_url": "https://api.github.com/users/k8s-ci-robot/orgs",
    "received_events_url": "https://api.github.com/users/k8s-ci-robot/received_events",
    "repos_url": "https://api.github.com/users/k8s-ci-robot/repos",
    "site_admin": false,
    "starred_url": "https://api.github.com/users/k8s-ci-robot/starred{/owner}{/repo}",
    "subscriptions_url": "https://api.github.com/users/k8s-ci-robot/subscriptions",
    "type": "User",
    "url": "https://api.github.com/users/k8s-ci-robot"
  },
  "comments": 13,
  "comments_url": "https://api.github.com/repos/kubernetes/kubernetes/issues/84595/comments",
  "created_at": "2019-10-31T13:03:34Z",
  "events_url": "https://api.github.com/repos/kubernetes/kubernetes/issues/84595/events",
  "html_url": "https://github.com/kubernetes/kubernetes/issues/84595",
  "id": 515416749,
  "labels": [
    {
      "color": "e11d21",
      "default": false,
      "description": "Categorizes issue or PR as related to a bug.",
      "id": 105146071,
      "name": "kind/bug",
      "node_id": "MDU6TGFiZWwxMDUxNDYwNzE=",
      "url": "https://api.github.com/repos/kubernetes/kubernetes/labels/kind/bug"
    },
    {
      "color": "d2b48c",
      "default": false,
      "description": "Categorizes an issue or PR as relevant to SIG Node.",
      "id": 173493665,
      "name": "sig/node",
      "node_id": "MDU6TGFiZWwxNzM0OTM2NjU=",
      "url": "https://api.github.com/repos/kubernetes/kubernetes/labels/sig/node"
    }
  ],
  "labels_url": "https://api.github.com/repos/kubernetes/kubernetes/issues/84595/labels{/name}",
  "locked": false,
  "milestone": null,
  "node_id": "MDU6SXNzdWU1MTU0MTY3NDk=",
  "number": 84595,
  "performed_via_github_app": null,
  "repository_url": "https://api.github.com/repos/kubernetes/kubernetes",
  "state": "closed",
  "title": "Errors returned from pod_container_manager_linux Destroy to remove a pods cgroup are swallowed",
  "updated_at": "2020-05-17T01:59:37Z",
  "url": "https://api.github.com/repos/kubernetes/kubernetes/issues/84595",
  "user": {
    "avatar_url": "https://avatars1.githubusercontent.com/u/6206907?v=4",
    "events_url": "https://api.github.com/users/mrobson/events{/privacy}",
    "followers_url": "https://api.github.com/users/mrobson/followers",
    "following_url": "https://api.github.com/users/mrobson/following{/other_user}",
    "gists_url": "https://api.github.com/users/mrobson/gists{/gist_id}",
    "gravatar_id": "",
    "html_url": "https://github.com/mrobson",
    "id": 6206907,
    "login": "mrobson",
    "node_id": "MDQ6VXNlcjYyMDY5MDc=",
    "organizations_url": "https://api.github.com/users/mrobson/orgs",
    "received_events_url": "https://api.github.com/users/mrobson/received_events",
    "repos_url": "https://api.github.com/users/mrobson/repos",
    "site_admin": false,
    "starred_url": "https://api.github.com/users/mrobson/starred{/owner}{/repo}",
    "subscriptions_url": "https://api.github.com/users/mrobson/subscriptions",
    "type": "User",
    "url": "https://api.github.com/users/mrobson"
  }
}